### ISSUE TITLE
How to upload a base64 encoded image as a file?

### ISSUE THREAD

**Issue by JamesJGoodwin** [form-data/form-data#463]
Hello. I have an image that is encoded in base64. How do I upload it with form-data without writing it to a file system and reading from it with `createReadStream()`? I have this code currently to make a valid uploadable file:

```javascript
fs.writeFileSync('image.png', img, { encoding: 'base64' })

const form = new FormData()
form.append('photo', fs.createReadStream('image.png'))
```

What I'm trying to achieve is to upload file without touching a filesystem. I tried various methods including `form.append('photo', Buffer.from(base64str, 'base64'))` but it's not working. And documentation doesn't say anything about uploading a base64 encoded image as a file.

Sorry to ask for a help in a bugtracker. I tried searching StackOverflow but none of the methods I found was working. Also, if there is a way to do what I'm trying to do then it's probably a good idea to include this in the docs.

**Comment by C-Saunders** [comment]
Have you tried something like this?
```ts
form.append('photo', Buffer.from(img, 'base64'), { filename: 'image.png' })
```

**Comment by JamesJGoodwin** [comment]
@C-Saunders yes I did. Server is not accepting it like this. I have created a repl for myself, but you can test it as well if you have any ideas: https://repl.it/@JamesJGoodwin/VigilantSparseDeletion

**Comment by castropaykode** [comment]
@abriginets Hello, did you find a way around this? I am in the same boat here

**Comment by mathias22osterhagen22** [comment]
Same issue, it seems that the file need to be write on disk first.

**Comment by mathias22osterhagen22** [comment]
I finally found a way around this:

**1.First**
If like me you were first downloading the file from somewhere else, pay attention to add `responseType: 'arraybuffer'` in `axios` options of the request. And make sure the document you download is valid, by opening it and checking it first.

**2.Second** 
Add your file to your formData with the `filename` and `contentType` options: 
```js
 formData.append('image', documentBytes, { filename: 'tmp_pdf.pdf', contentType: 'application/pdf' });
```

**3.Axios**
If you are uploading the document with `axios`, do not forget to add the `formData.getHeaders()` to your actual axios headers: 

```js
 {
      headers: {
          ...formData.getHeaders(),
      }
  }
```

This made the tricks for me
